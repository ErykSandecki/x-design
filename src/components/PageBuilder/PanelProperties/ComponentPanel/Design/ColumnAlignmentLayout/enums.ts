export enum PopoverItem {
  fixed = 0,
}
