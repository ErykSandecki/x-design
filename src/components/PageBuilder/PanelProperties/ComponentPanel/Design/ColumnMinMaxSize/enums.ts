export enum PopoverItem {
  currentValue = 0,
  auto = 1,
  unit = 2,
  removeScore = 3,
}
