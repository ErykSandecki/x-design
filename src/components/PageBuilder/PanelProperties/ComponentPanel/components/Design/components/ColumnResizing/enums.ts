export enum PopoverItem {
  auto = 1,
  fixed = 0,
  maxScore = 4,
  minScore = 3,
  unit = 2,
}
