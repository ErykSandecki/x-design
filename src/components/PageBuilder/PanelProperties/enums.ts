export enum Tab {
  design = 'design',
  prototype = 'prototype',
}
