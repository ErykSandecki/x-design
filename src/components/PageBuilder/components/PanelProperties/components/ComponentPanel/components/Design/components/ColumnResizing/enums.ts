export enum PopoverItem {
  auto = 1,
  fixed = 0,
  unit = 2,
}
