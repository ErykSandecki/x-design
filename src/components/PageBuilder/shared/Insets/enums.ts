export enum InsetMode {
  individual = 'individual',
  merged = 'merged',
}
