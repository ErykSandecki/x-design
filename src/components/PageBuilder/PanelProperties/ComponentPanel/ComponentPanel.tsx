import { FC } from 'react';

// components
import Design from './Design/Design';

// types
import { Tab } from '../enums';

export type TMainPanel = {
  activeTab: Tab;
  width: number;
};

const ComponentPanel: FC<TMainPanel> = ({ activeTab, width }) => {
  if (activeTab === Tab.design) {
    return <Design width={width} />;
  }

  return <></>;
};

export default ComponentPanel;
