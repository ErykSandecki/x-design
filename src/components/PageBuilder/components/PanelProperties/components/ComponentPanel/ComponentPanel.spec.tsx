import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import ComponentPanel from './ComponentPanel';

// mocks
import {
  elementMock,
  elementDynamicDataMock,
  elementStaticDataMock,
  pageBuilderStateMock,
  selectedElementMock,
} from 'test/mocks/reducer/pageBuilderMock';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { Tab } from '../../enums';

const currentPage = pageBuilderStateMock[PAGE_BUILDER].pages['0'];
const stateMock = {
  ...pageBuilderStateMock,
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...currentPage,
        elements: {
          allData: {
            ['-1']: {
              ...currentPage.elements.allData['-1'],
              children: [elementMock.id],
            },
            [elementMock.id]: {
              ...elementMock,
            },
          },
          dynamicData: {
            ['-1']: {
              ...currentPage.elements.dynamicData['-1'],
              children: [elementDynamicDataMock.id],
            },
            [elementDynamicDataMock.id]: elementDynamicDataMock,
          },
          staticData: {
            ['-1']: {
              ...currentPage.elements.staticData['-1'],
              children: [elementStaticDataMock.id],
            },
            [elementStaticDataMock.id]: {
              ...elementStaticDataMock,
            },
          },
        },
        selectedElements: [selectedElementMock],
      },
    },
  },
};

describe('ComponentPanel snapshots', () => {
  it('should render ComponentPanel', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ComponentPanel activeTab={Tab.design} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render proptype section', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ComponentPanel activeTab={Tab.prototype} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
