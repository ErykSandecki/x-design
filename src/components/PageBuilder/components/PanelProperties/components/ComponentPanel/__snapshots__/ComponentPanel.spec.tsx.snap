// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentPanel snapshots should render ComponentPanel 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp Section Section--light"
    data-test-section=""
  >
    <div
      class="css-1jnxvbp SectionColumn SectionColumn--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__labels SectionColumn__labels--light"
        style="width: calc(100% - 48px - 2.5px - 8px);"
      />
      <div
        class="SectionColumn__wrapper SectionColumn__wrapper--light"
      >
        <div
          class="SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
          style="width: calc(100% - 48px - 2.5px - 8px);"
        >
          <small
            class="css-1jnxvbp Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
            data-test-text=""
            style="text-align: inherit;"
          >
            Frame
          </small>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        >
          <div
            class="css-1jnxvbp ButtonIcon ButtonIcon--light"
            data-test-button-icon=""
          >
            <div
              class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="html-tag"
              height="14"
              name="HtmlTag"
              width="14"
            />
          </div>
          <div
            class="css-1jnxvbp ButtonIcon ButtonIcon--light"
            data-test-button-icon=""
          >
            <div
              class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="component"
              height="14"
              name="Component"
              width="14"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-1jnxvbp Section Section--light"
    data-test-section=""
  >
    <div
      class="Section__header Section__header--light"
    >
      <small
        class="css-1jnxvbp Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Position
      </small>
      <div
        class="Section__buttons Section__buttons--light"
      />
    </div>
    <div
      class="css-1jnxvbp SectionColumn SectionColumn--light SectionColumn--withMargin SectionColumn--withMargin--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__labels SectionColumn__labels--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <small
          class="css-1jnxvbp SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Alignment
        </small>
      </div>
      <div
        class="SectionColumn__wrapper SectionColumn__wrapper--light"
      >
        <div
          class="SectionColumn__content SectionColumn__content--light SectionColumn__content--twoInputs SectionColumn__content--twoInputs--light"
          style="width: calc(100% - 24px - 0px - 8px);"
        >
          <div
            class="css-1jnxvbp ButtonGroup ButtonGroup--light"
            data-test-button-group="horizontal-alignment"
          >
            <div
              class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
              data-test-button-group-input="align-horizontal-left"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="align-horizontal-left"
                height="12"
                name="AlignHorizontalLeft"
                width="12"
              />
            </div>
            <div
              class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
              data-test-button-group-input="align-horizontal-center"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="align-horizontal-center"
                height="12"
                name="AlignHorizontalCenter"
                width="12"
              />
            </div>
            <div
              class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
              data-test-button-group-input="align-horizontal-right"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="align-horizontal-right"
                height="12"
                name="AlignHorizontalRight"
                width="12"
              />
            </div>
          </div>
          <div
            class="css-1jnxvbp ButtonGroup ButtonGroup--light"
            data-test-button-group="vertical-alignment"
          >
            <div
              class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
              data-test-button-group-input="align-vertical-top"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="align-vertical-top"
                height="12"
                name="AlignVerticalTop"
                width="12"
              />
            </div>
            <div
              class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
              data-test-button-group-input="align-vertical-center"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="align-vertical-center"
                height="12"
                name="AlignVerticalCenter"
                width="12"
              />
            </div>
            <div
              class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
              data-test-button-group-input="align-vertical-bottom"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="align-vertical-bottom"
                height="12"
                name="AlignVerticalBottom"
                width="12"
              />
            </div>
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp SectionColumn SectionColumn--light SectionColumn--withMargin SectionColumn--withMargin--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__labels SectionColumn__labels--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <small
          class="css-1jnxvbp SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Position
        </small>
      </div>
      <div
        class="SectionColumn__wrapper SectionColumn__wrapper--light"
      >
        <div
          class="SectionColumn__content SectionColumn__content--light SectionColumn__content--twoInputs SectionColumn__content--twoInputs--light"
          style="width: calc(100% - 24px - 0px - 8px);"
        >
          <div
            class="css-1jnxvbp TextField TextField--light TextField--fullWidth TextField--fullWidth--light"
            data-test-text-field="x"
          >
            <div
              class="ScrubbableInput ScrubbableInput--light"
              data-test-scrubbable-input="x"
            >
              <small
                class="css-1jnxvbp Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                data-test-text=""
                style="text-align: inherit;"
              >
                X
              </small>
            </div>
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="x"
              maxlength="6"
              type="number"
              value="0"
            />
          </div>
          <div
            class="css-1jnxvbp TextField TextField--light TextField--fullWidth TextField--fullWidth--light"
            data-test-text-field="y"
          >
            <div
              class="ScrubbableInput ScrubbableInput--light"
              data-test-scrubbable-input="y"
            >
              <small
                class="css-1jnxvbp Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                data-test-text=""
                style="text-align: inherit;"
              >
                Y
              </small>
            </div>
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="y"
              maxlength="6"
              type="number"
              value="0"
            />
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp SectionColumn SectionColumn--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__labels SectionColumn__labels--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <small
          class="css-1jnxvbp SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Rotation
        </small>
      </div>
      <div
        class="SectionColumn__wrapper SectionColumn__wrapper--light"
      >
        <div
          class="SectionColumn__content SectionColumn__content--light SectionColumn__content--twoInputs SectionColumn__content--twoInputs--light"
          style="width: calc(100% - 24px - 0px - 8px);"
        >
          <div
            class="css-1jnxvbp TextField TextField--light TextField--fullWidth TextField--fullWidth--light"
            data-test-text-field="angle"
          >
            <div
              class="ScrubbableInput ScrubbableInput--light"
              data-test-scrubbable-input="angle"
            >
              <div
                class="css-1jnxvbp ScrubbableInput__icon ScrubbableInput__icon--light Icon Icon--light Icon--neutral2 Icon--neutral2--light"
                data-test-icon="protractor"
                height="16"
                name="Protractor"
                width="16"
              />
            </div>
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="angle"
              maxlength="6"
              type="number"
              value="0"
            />
          </div>
          <div
            class="css-1jnxvbp ButtonGroup ButtonGroup--light ButtonGroup--fullWidth ButtonGroup--fullWidth--light"
            data-test-button-group="layout-position"
          >
            <div
              class="ButtonGroup__button ButtonGroup__button--light"
              data-test-button-group-input="toggle-rotate"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="toggle-rotate"
                height="12"
                name="ToggleRotate"
                width="12"
              />
            </div>
            <div
              class="ButtonGroup__button ButtonGroup__button--light"
              data-test-button-group-input="flip-horizontal"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="flip-horizontal"
                height="12"
                name="FlipHorizontal"
                width="12"
              />
            </div>
            <div
              class="ButtonGroup__button ButtonGroup__button--light"
              data-test-button-group-input="flip-vertical"
            >
              <div
                class="css-1jnxvbp ButtonGroup__icon ButtonGroup__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="flip-vertical"
                height="12"
                name="FlipVertical"
                width="12"
              />
            </div>
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        />
      </div>
    </div>
  </div>
  <div
    class="css-1jnxvbp Section Section--light"
    data-test-section=""
  >
    <div
      class="Section__header Section__header--light"
    >
      <small
        class="css-1jnxvbp Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Layout
      </small>
      <div
        class="Section__buttons Section__buttons--light"
      >
        <div
          class="css-1jnxvbp ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="fit-layout"
            height="14"
            name="FitLayout"
            width="14"
          />
        </div>
        <div
          class="css-1jnxvbp ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="auto-layout"
            height="14"
            name="AutoLayout"
            width="14"
          />
        </div>
      </div>
    </div>
    <div
      class="css-1jnxvbp SectionColumn SectionColumn--light SectionColumn--withMargin SectionColumn--withMargin--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__labels SectionColumn__labels--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <small
          class="css-1jnxvbp SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Flow
        </small>
      </div>
      <div
        class="SectionColumn__wrapper SectionColumn__wrapper--light"
      >
        <div
          class="SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
          style="width: calc(100% - 24px - 0px - 8px);"
        >
          <div
            class="ToggleButtonGroup ToggleButtonGroup--light ToggleButtonGroup--fullWidth ToggleButtonGroup--fullWidth--light"
            data-test-toggle-button-group="flow"
          >
            <button
              class="ToggleButton ToggleButton--light ToggleButton--selected ToggleButton--selected--light"
              data-test-toggle-button="0"
              type="button"
            >
              <div
                class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="flow-default"
                height="12"
                name="FlowDefault"
                width="12"
              />
            </button>
            <button
              class="ToggleButton ToggleButton--light"
              data-test-toggle-button="1"
              type="button"
            >
              <div
                class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="flow-vertical"
                height="12"
                name="FlowVertical"
                width="12"
              />
            </button>
            <button
              class="ToggleButton ToggleButton--light"
              data-test-toggle-button="2"
              type="button"
            >
              <div
                class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="flow-horizontal"
                height="12"
                name="FlowHorizontal"
                width="12"
              />
            </button>
            <button
              class="ToggleButton ToggleButton--light"
              data-test-toggle-button="3"
              type="button"
            >
              <div
                class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                data-test-icon="flow-grid"
                height="12"
                name="FlowGrid"
                width="12"
              />
            </button>
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp SectionColumn SectionColumn--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__labels SectionColumn__labels--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <small
          class="css-1jnxvbp SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Resizing
        </small>
      </div>
      <div
        class="SectionColumn__wrapper SectionColumn__wrapper--light"
      >
        <div
          class="SectionColumn__content SectionColumn__content--light SectionColumn__content--twoInputs SectionColumn__content--twoInputs--light"
          style="width: calc(100% - 24px - 0px - 8px);"
        >
          <div
            class="css-1jnxvbp TextField TextField--light TextField--fullWidth TextField--fullWidth--light"
            data-test-text-field="width"
          >
            <div
              class="ScrubbableInput ScrubbableInput--light"
              data-test-scrubbable-input="width"
            >
              <small
                class="css-1jnxvbp Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                data-test-text=""
                style="text-align: inherit;"
              >
                W
              </small>
            </div>
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="width"
              maxlength="6"
              type="number"
              value="100"
            />
            <div
              class="css-1jnxvbp TextFieldPopover TextFieldPopover--light"
              data-test-box=""
            >
              <div
                class="css-1jnxvbp TextFieldPopover__icon TextFieldPopover__icon--light TextField__icon TextField__icon--light Icon Icon--light Icon--clickable Icon--clickable--light"
                data-test-icon="variant"
                height="12"
                name="Variant"
                width="12"
              />
              <div
                class="Popover Popover--light"
                data-test-popover="popover"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="0"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light PopoverItem__checkIcon--selected PopoverItem__checkIcon--selected--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="fixed-width"
                    height="12"
                    name="FixedWidth"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Fixed width
                  </small>
                </div>
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="1"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="auto-width"
                    height="12"
                    name="AutoWidth"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Auto width
                  </small>
                </div>
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="2"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="percentage"
                    height="12"
                    name="Percentage"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Percentage
                  </small>
                </div>
                <div
                  class="PopoverSeparator PopoverSeparator--light"
                />
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="3"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="min-width"
                    height="12"
                    name="MinWidth"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Percentage
                  </small>
                </div>
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="4"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="max-width"
                    height="12"
                    name="MaxWidth"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Add min width...
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-1jnxvbp TextField TextField--light TextField--fullWidth TextField--fullWidth--light"
            data-test-text-field="height"
          >
            <div
              class="ScrubbableInput ScrubbableInput--light"
              data-test-scrubbable-input="height"
            >
              <small
                class="css-1jnxvbp Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                data-test-text=""
                style="text-align: inherit;"
              >
                H
              </small>
            </div>
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="height"
              maxlength="6"
              type="number"
              value="100"
            />
            <div
              class="css-1jnxvbp TextFieldPopover TextFieldPopover--light"
              data-test-box=""
            >
              <div
                class="css-1jnxvbp TextFieldPopover__icon TextFieldPopover__icon--light TextField__icon TextField__icon--light Icon Icon--light Icon--clickable Icon--clickable--light"
                data-test-icon="variant"
                height="12"
                name="Variant"
                width="12"
              />
              <div
                class="Popover Popover--light"
                data-test-popover="popover"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="0"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light PopoverItem__checkIcon--selected PopoverItem__checkIcon--selected--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="fixed-height"
                    height="12"
                    name="FixedHeight"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Fixed height
                  </small>
                </div>
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="1"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="auto-height"
                    height="12"
                    name="AutoHeight"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Auto height
                  </small>
                </div>
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="2"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="percentage"
                    height="12"
                    name="Percentage"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Percentage
                  </small>
                </div>
                <div
                  class="PopoverSeparator PopoverSeparator--light"
                />
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="3"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="min-height"
                    height="12"
                    name="MinHeight"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Percentage
                  </small>
                </div>
                <div
                  class="PopoverItem PopoverItem--light"
                  data-test-popover-item="4"
                >
                  <div
                    class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="check"
                    height="12"
                    name="Check"
                    width="12"
                  />
                  <div
                    class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="max-height"
                    height="12"
                    name="MaxHeight"
                    width="12"
                  />
                  <small
                    class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Add min height...
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ComponentPanel snapshots should render proptype section 1`] = `<DocumentFragment />`;
